build.fsx

```diff
@@ -313,3 +312,153 @@ pipeline "Init" {
     }
     runIfOnlySpecified true
 }
+
+type GithubRelease =
+    { Version: string
+      Title: string
+      Date: DateTime
+      Exists: bool
+      Draft: string }
+
+let mkGithubRelease (v: SemanticVersion, d: DateTime, cd: ChangelogData option) =
+    match cd with
+    | None -> failwith "Each Fantomas release is expected to have at least one section."
+    | Some cd ->
+        let version = $"{v.Major}.{v.Minor}.{v.Patch}"
+        let title =
+            let month = d.ToString("MMMM")
+            let day = d.Day.Ordinalize()
+            $"{month} {day} Release"
+
+        let prefixedVersion = $"v{version}"
+        let exists =
+            let cmdResult =
+                Cli
+                    .Wrap("gh")
+                    .WithArguments($"release view {prefixedVersion}")
+                    .WithValidation(CommandResultValidation.None)
+                    .ExecuteAsync()
+                    .Task.Result
+            cmdResult.ExitCode = 0
+
+        let sections =
+            [ "Added", cd.Added
+              "Changed", cd.Changed
+              "Fixed", cd.Fixed
+              "Deprecated", cd.Deprecated
+              "Removed", cd.Removed
+              "Security", cd.Security
+              yield! (Map.toList cd.Custom) ]
+            |> List.choose (fun (header, lines) ->
+                if lines.IsEmpty then
+                    None
+                else
+                    lines
+                    |> List.map (fun line -> line.TrimStart())
+                    |> String.concat "\n"
+                    |> sprintf "### %s\n%s" header
+                    |> Some)
+            |> String.concat "\n\n"
+
+        let draft =
+            $"""# {version}
+
+{sections}"""
+
+        { Version = version
+          Title = title
+          Date = d
+          Exists = exists
+          Draft = draft }
+
+let getReleaseNotes currentRelease lastRelease =
+    let date = lastRelease.Date.ToString("yyyy-MM-dd")
+    let authorMsg =
+        let authors =
+            Cli
+                .Wrap("gh")
+                .WithArguments(
+                    $"pr list -S \"state:closed base:main closed:>{date} -author:app/robot\" --json author --jq \".[].author.login\""
```

Comment:
Mmh, I think that could miss authors with PRs closed after the last release but still on the same day as the last release.
Not really critical but something to be aware of.

Reply:
Yeah, this is a bit of a trade-off. When using `closed:>={date}` authors would be included twice if we release on the same day.

Reply:
Yes, it's okay for me. Just wanted to point out that we should be careful if we have such a day.